src/SWP/Bundle/CoreBundle/Twig/ArticleBodyExtension.php

```diff
@@ -49,9 +49,69 @@ public function getFunctions(): array
     {
         return [
             new TwigFunction('setRendition', [$this, 'setRendition']),
+            new TwigFunction('setInlineAds', [$this, 'setInlineAds']),
         ];
     }
 
+    public function setInlineAds(string $body): array
+    {
+        # split the body into paragraphs
+        $paragraphs = explode("</p>", $body);
```

Comment:
Don't we loose the closing p-tags?

`explode("</p>", '<h1>headline</h1><p>paragraph one.</p><p>paragraphtwo.</p>');`

Results in
```
Array
(
    [0] => <h1>headline</h1><p>paragraph one.
    [1] => <p>paragraphtwo.
    [2] => 
)
```

Reply:
i've added the delimiter back. but this also handled in the template using the raw filter.